lib/APMSpan.js

```diff
@@ -65,7 +73,43 @@ class APMSpan {
             nodeType = 'Consumer';
         }
 
-        this[SPAN_CONTEXT].getTrace().addNode(nodeType, this, consumerCorrelationId);
+        this[SPAN_CONTEXT].getTrace().addNode(nodeType, this, consumerCorrelationId ? [{
+            value: consumerCorrelationId,
+            scope: 'Interaction',
+        }] : undefined);
+    }
+
+    _initFollowsFromSpanContext(parentReferences) {
+        function findRoot(span) {
+            let rootSpan = span;
+            while (rootSpan.getParentId()) {
+                const node = span.getTrace().findNode(rootSpan.getParentId());
+                if (node) {
+                    rootSpan = node.getSpan();
+                } else {
+                    break;
+                }
+            }
+
+            return rootSpan;
+        }
+
+        const id = utils.generateSpanId();
+        let traceId = null;
+
+        const correlationIds = [];
+        for (let i = 0; i < parentReferences.length; i += 1) {
+            const parentReference = parentReferences[i] instanceof APMSpan ? parentReferences[i].context() : parentReferences[i];
```

Comment:
References are not a list of Span or SpanContext objects, they are of type opentracing.Reference, e.g. https://github.com/opentracing/opentracing-javascript/blob/master/test/opentracing_api.js#L65